src/mavlink_server.cpp

```diff
@@ -303,3 +529,48 @@ void MavlinkServer::stop()
     if (_timeout_handler > 0)
         Mainloop::get_mainloop()->del_timeout(_timeout_handler);
 }
+
+int MavlinkServer::addCameraComponent(CameraComponent *camComp)
+{
+    log_debug("%s", __func__);
+    int id = MAV_COMP_ID_CAMERA;
+    while (id < MAV_COMP_ID_CAMERA6 + 1) {
+        if (compIdToObj.find(id) == compIdToObj.end()) {
+            compIdToObj.insert(std::make_pair(id, camComp));
+            break;
+        }
+        id++;
+    }
+
+    return id;
+}
+
+void MavlinkServer::removeCameraComponent(CameraComponent *camComp)
+{
+    log_debug("%s", __func__);
+
+    if (!camComp)
+        return;
+
+    for (std::map<int, CameraComponent *>::iterator it = compIdToObj.begin();
+         it != compIdToObj.end(); it++) {
+        if ((it->second) == camComp) {
+            compIdToObj.erase(it);
+            break;
+        }
+    }
+
+    return;
+}
+
+CameraComponent *MavlinkServer::getCameraComponent(int compID)
+{
+    if (compID < MAV_COMP_ID_CAMERA || compID > MAV_COMP_ID_CAMERA6)
```

Comment:
Well, anyway I don't like the idea of having a limit on the number of available cameras in the protocol. Even if it is big enough. But that's something to discuss with the dronecode community. 
And if I install camera-streaming-daemon in my product and I connect more than 6 usb cameras in it, it won't work. It is good to document that the camera-streaming-daemon will support only 6 cameras and inform users on how to change this if needed.